FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_Sockets.c

```diff
@@ -299,6 +304,15 @@ Socket_t xReturn;
 		}
 		else
 		{
+			if( xProtocol == FREERTOS_IPPROTO_UDP )
```

Comment:
Rather than add these new branches, we could replace the various malloc calls in this module with a macro (e.g. pvPortSocketsMalloc). The macro could accept whatever additional call site-specific parameters are required for stats. Would there be disadvantages with that approach?

Reply:
@dcgaws: I have thought of doing so, but the macro `pvPortMallocSocket()` only takes one parameter: the size. I think we'de like to see in the logging what type of socket was created, UDP or TCP.
Likewise, `pvPortMallocLarge()` only takes a parameter `uxSize`. That function creates the stream buffers of TCP.
A socket-set is allocated with the standard FreeRTOS function `pvPortMalloc()`, which is too generic.

The new macro's like `iptraceMEM_STATS_CREATE()` get their default values in `IPTraceMacroDefaults.h`. Users can define them for their own use ( in `FreeRTOSIPConfig.h` ), or use the new modules.